src/main/java/io/temporal/samples/hello/HelloPeriodic.java

```diff
@@ -106,31 +133,45 @@
             GreetingActivities.class,
             ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(10)).build());
 
-    // Create our workflow stub that can be used to continue this workflow as a new run
-    // We use this because we want each workflow execution to be a separate workflow invocation
-    private final GreetingWorkflow continueAsNew =
-        Workflow.newContinueAsNewStub(GreetingWorkflow.class);
+    private boolean exitRequested = false;
+
+    @Override
+    public void requestExit() {
+      exitRequested = true;
+    }
 
     @Override
-    public void greetPeriodically(String name) {
-
-      // Loop as many times as defined in CONTINUE_AS_NEW_FREQUENCY
-      /*
-       * Here we define how many times we want to execute our workflow activity periodically.
-       * The value is set to 10 for the sake of the example. In real applications it would make
-       * more sense to set this to a higher value that matches a business cycle (for example once
-       * per 24 hours, etc).
-       */
-      int CONTINUE_AS_NEW_FREQUENCY = 10;
-
-      for (int i = 0; i < CONTINUE_AS_NEW_FREQUENCY; i++) {
-        // execute our activity method and sleep for a random amount of time
-        int delayMillis = random.nextInt(10000);
-        activities.greet("Hello " + name + "! Sleeping for " + delayMillis + " milliseconds.");
-        Workflow.sleep(delayMillis);
+    public boolean greetPeriodically(String name) {
+
+      for (int i = 0; i < SingleWorkflowRunIterations; i++) {
+
+        // Compute the timing of the next iteration:
+        int delayMillis = (ScheduleConfig.PeriodTargetSecs * 1000)
+                          + random.nextInt(ScheduleConfig.PeriodVariationSecs * 1000)
+                          - (ScheduleConfig.PeriodVariationSecs * 500);
+
+        // Perform some useful work. In this example, we execute a greeting activity:
+        activities.greet("Hello " + name + "!"
+                       + " I will sleep for " + delayMillis + " milliseconds and then I will greet you again.");
+
+        // Sleep for the required time period or until an exit signal is received:
+        Workflow.await(Duration.ofMillis(delayMillis), () -> exitRequested);
+
+        if (exitRequested) {
+          activities.greet("Hello " + name + "!"
+                         + " It was requested to quit the periodic greetings, so this the last one.");
+          return true;
+        }
       }
-      // Stop execution of the current workflow and start a new execution
+
+      // Create a workflow stub that will be used to continue this workflow as a new run:
+      // (see the comment for 'SingleWorkflowRunIterations' for details)
+      GreetingWorkflow continueAsNew = Workflow.newContinueAsNewStub(GreetingWorkflow.class);
+
+      // Request that the new run will be invoked by the Temporal system:
       continueAsNew.greetPeriodically(name);
+
+      return false;
```

Comment:
Why do we return boolean here? It is useless as it will only return the value of the last one to the caller which is always true.

Reply:
✔️